modules/javafx.controls/src/main/java/javafx/scene/chart/BarChart.java

```diff
@@ -223,14 +223,18 @@ public BarChart(@NamedArg("xAxis") Axis<X> xAxis, @NamedArg("yAxis") Axis<Y> yAx
                 // by looking at the concatenation of the data of all series, skipping duplicate categories.
                 categoryIndex = 0;
                 var uniqueCategories = new HashSet<String>();
-                for (Map<String, Data<X,Y>> catMap : seriesCategoryMap.values()) {
+                for (var entry : seriesCategoryMap.entrySet()) {
+                    Series s = entry.getKey();
```

Comment:
would it make sense to add a comment explaining the reasoning for this more complex code?

Reply:
I've extended the code comment.

Reply:
thank you